### PR TITLE
Fixing Web Crashes

### PR DESCRIPTION
**Pull request by erickzanardo**
# Description

This PR solves the weird bug that made games on web crash catastrophically on web.

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue) I don't think this is a breaking change, let me know what you think, because of the const contructor.

# Checklist:

If something is unclear, please submit the PR anyways and ask about what you thought was unclear.

- [x] This branch is based on `develop`
- [x] This PR is targeted to merge into `develop` (not `master`)
- [x] I have added an entry under `[next]` in `CHANGELOG.md`
- [x] I have formatted my code with `flutter format`
- [x] I have added examples for new features in `doc/examples`
- [x] The continuous integration (CI) is passing

### CODE REVIEW

File: lib/memory_cache.dart [hunk 1]
```diff
@@ -0,0 +1,27 @@
+/// Simple class to cache values on the cache
+///
+class MemoryCache<K, V> {
+  final Map<K, V> _cache = {};
+  final List<K> _addedOrder = [];
```
Comment (spydon) [inline lib/memory_cache.dart:5]: Can't you just use an ordered map instead?

File: lib/text_config.dart [hunk 2]
```diff
@@ -52,10 +53,13 @@ class TextConfig {
...
   final TextDirection textDirection;
 
+  final MemoryCache _textPainterCache =
+      MemoryCache<String, material.TextPainter>();
+
   /// Creates a constant [TextConfig] with sensible defaults.
   ///
   /// Every parameter can be specified.
-  const TextConfig({
+  TextConfig({
```
Comment (spydon) [inline lib/text_config.dart:62]: How come they can't be const anymore?

Comment (renancaraujo) [reply lib/text_config.dart:62]: Now `_textPainterCache` contains some non const collections.

File: lib/memory_cache.dart (hunk 1 above)
Comment (erickzanardo) [reply lib/memory_cache.dart]: Good call, felt like a noob now.

Made the change.

File: lib/text_config.dart (hunk 2 above)
Comment (erickzanardo) [reply lib/text_config.dart:62]: Exactly!

File: lib/memory_cache.dart [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import 'dart:collection';
+
+/// Simple class to cache values on the cache
+///
+class MemoryCache<K, V> {
```
Comment (luanpotter) [inline lib/memory_cache.dart:5]: should we refactor our file and audio caches to use this somehow?

File: lib/components/text_box_component.dart [hunk 4]
```diff
@@ -46,11 +46,13 @@ class TextBoxComponent extends PositionComponent with Resizable {
...
 
-  TextBoxComponent(String text,
-      {TextConfig config = const TextConfig(),
-      TextBoxConfig boxConfig = const TextBoxConfig()}) {
-    _boxConfig = boxConfig;
-    _config = config;
+  TextBoxComponent(
+    String text, {
+    TextConfig config,
+    TextBoxConfig boxConfig,
```
Comment (luanpotter) [inline lib/components/text_box_component.dart:52]: nice!!!

File: lib/memory_cache.dart (hunk 3 above)
Comment (erickzanardo) [reply lib/memory_cache.dart:5]: I don’t think so, at least not without changing this class a little, on audio and image cache, we don’t automatically clear the cache, everything is manual there.

We could change this class to have an infinity cache and then use it there. But I wouldn’t do this on this PR so we don’t mix stuff

Comment (luanpotter) [reply lib/memory_cache.dart:5]: yeah totally, I meant as a followup. def. not a part of this PR

**Pull request merged**